### DISCUSSION TITLE
LXQt missing icons

### DISCUSSION THREAD

**Discussion by bdubbs** [lxqt/lxqt#2560] [Q&A]
I am trying to present build instructions for LXQt in Beyond Linux From Scratch. https://www.linuxfromscratch.org/blfs/view/svn/lxqt/lxqt-desktop.html

Everything builds but 90% of the icons are missing in places like the Configuration Center.  I've tried several things including several different Icon Theme/LXQt Theme combinations but nothing seems to work.  I've also built KF6/Plasma and xfce and all the icons seem to be present for those desktop environments.  

What am I missing?

**Comment by stefonarch** [comment]
Is any icon theme set in `~/.config/lxqt/lxqt.conf `?

**Reply by bdubbs** [reply]
Yes.  For instance right now I have:

```
[General]
__userfile__=true
icon_follow_color_scheme=true
icon_theme=oxygen
theme=Clearlooks
tool_button_style=ToolButtonTextBesideIcon

```

What theme/icon_theme would you suggest?

**Reply by tsujan** [reply]
> What theme/icon_theme would you suggest?

It's hard to suggest an icon theme because there are many good ones. Oxygen is OK.

Your problem doesn't seem to be about the theme. Are icons missing in other Qt6 apps too? Do you run another platform plugin (like qt6ct) in parallel to that of LXQt?

**Reply by tsujan** [reply]
And more basically, is `XDG_CURRENT_DESKTOP` set to `LXQt` in your session?

**Reply by Butterfly** [reply]
Hey, Bruce! You're messing with my head this morning. Am not used to seeing you ask questions, lol. Is there any chance that `apt-file` might help? As an example, I just tried this (plucked from this current thread):

```
$ apt-file find lxqt/lxqt.conf
lxqt-branding-debian: /etc/xdg/lxqt/lxqt.conf
lxqt-session: /usr/share/lxqt/lxqt.conf
```

Just thinking out loud because it would possibly show where something might have gone missing between the original archived package and that package's expanded installation. Another thought is maybe it would hint at a location that contains conflicting variables or suchly.

DISCLAIMER: Obviously based on the two query results in my example, `apt-file` doesn't cover personalizations, but it could still hint at preferred default paths for those personalizations.

**Reply by bdubbs** [reply]
My system is pretty bare right now.  I don't have a good non-Qt6 app with icons to test, but I can probably build one.  Can you suggest a candidate?  I can run konsole and the icons in the settings menu are present.

I normally do not use a file manager, but the icons in pcmanfm-qt *are* present.  The problem is that most small icons in the Configuration Center and other menus from the panel are missing.  For instance, in  Configuration Center I have icons for 'Brightness' and 'Openbox Settings' but no others.  Let me see if I can put a screen image here.

**Reply by stefonarch** [reply]
You _have_ installed `lxqt-menu-data` ? It shouldn't be possible to build without and it's mandatory  for the category icons in the application menu. 

Oops, obsolete ;)

**Comment by bdubbs** [comment]
![screen](https://github.com/lxqt/lxqt/assets/9353443/79d0d3f5-5ce7-4092-96e8-50e5d040ffa6)

**Comment by bdubbs** [comment]
Also let me note that 'export XDG_CURRENT_DESKTOP=LXQt' before running startx made no difference.

**Reply by tsujan** [reply]
`XDG_CURRENT_DESKTOP` should always be `LXQt` under LXQt. The icons of pcmanfm-qt are OK in your screenshot too.

Under an LXQt session, icons are handled by `lxqt-qtplugin`, which installs its plugin as `/usr/lib/qt6/plugins/platformthemes/libqtlxqt.so` (depending on the distro).

Something is wrong about lxqt-panel and probably lxqt-config in your case. With the current info, I can't guess what it may be.

**Comment by bdubbs** [comment]
That's it!  I built, but did not install lxqt-qtplugin properly.  Now all icons show properly.  Thank you so much.
